[vllm] [2.12 regression][multimodal] Qwen2-VL multi-image generation diverges from HF reference in test_case43

## Summary

Under torch 2.12.0 + triton 3.7.0, vLLM's `test_multi_image_models[qwen2_vl-test_case43]` fails with an output divergence between the HF reference and vLLM:

> `AssertionError: Test1:`
>
> Matched tokens (divergence point is after "a large"): `[785, 1156, 2168, 4933, 264, 2936, 1841, 448, 264, 2518, 4004, 323, 4158, 11931, 11, 9099, 304, 4065, 315, 264, 2518, 4752, 448, 264, 3460]`
>
> hf:   `'The first image shows a stop sign ... placed in front of a red building with a large window. ... The second image features a beautiful view of a cherry blossom tree ...`
>
> vllm: `'The first image shows a stop sign ... placed in front of a red building with a large archway. ... The second image features a beautiful cherry blossom tree ...`

Passes on torch 2.11. Blocking the torch 2.12 upgrade for vLLM (vllm-project/vllm#40077).

## Environment

- `torch`: 2.12.0+cu130 (test channel)
- `triton`: 3.7.0
- CUDA: 13.0 / Driver: 570.133.20
- Python: 3.12.13
- GPU: NVIDIA L4 (23 GiB)
- Model: `Qwen/Qwen2-VL-2B-Instruct`

## Reproduction

Failing test ID:

```
tests/models/multimodal/generation/test_common.py::test_multi_image_models[qwen2_vl-test_case43]
```

Test config: multi-image (stop_sign + cherry_blossom assets), size factors (0.25, 0.2, 0.15), `max_tokens=128`, `num_logprobs=5`, eager distributed.

## Reproducibility on torch 2.12 branch (3/3 days)

- 2026-04-20: https://buildkite.com/vllm/ci/builds/62138#019dab36-87e5-41f9-83d2-788681cea4de
- 2026-04-21: https://buildkite.com/vllm/ci/builds/62232#019dadac-e45c-443f-8d8b-6855ff38a476
- 2026-04-22: https://buildkite.com/vllm/ci/builds/62495#019db523-ef57-470c-a56b-332a1b7243c6

Passes on same-day main builds (torch 2.11):

- 2026-04-21 nightly: https://buildkite.com/vllm/ci/builds/62254
- 2026-04-21 daily: https://buildkite.com/vllm/ci/builds/62383
- 2026-04-22 nightly: https://buildkite.com/vllm/ci/builds/62456

## Diagnosis request

Output tokens diverge after a deterministic shared prefix across three runs — consistent with a kernel-level numerical change (matmul / softmax / attention / rotary), plausibly from the triton 3.7 bundled in torch 2.12. Is this an intentional numerical behavior change or a regression?

## Links

- vLLM PR: vllm-project/vllm#40077
- Umbrella: pytorch/pytorch#180899


cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

--- a/tests/models/multimodal/generation/test_common.py
+++ b/tests/models/multimodal/generation/test_common.py
@@ -921,6 +921,7 @@ VLM_TEST_SETTINGS = {
         multi_image_prompt="Picture 1: <vlm_image>\nPicture 2: <vlm_image>\nDescribe these two images with one paragraph respectively.",  # noqa: E501
         max_model_len=4096,
         max_num_seqs=2,
+        num_logprobs=10,
         auto_cls=AutoModelForImageTextToText,
         vllm_output_post_proc=model_utils.qwen2_vllm_to_hf_output,
         image_size_factors=[(0.25,), (0.25, 0.25, 0.25), (0.25, 0.2, 0.15)],

